temporalio/lib/temporalio/cancellation.rb

```diff
@@ -18,7 +18,7 @@ class Cancellation
     def initialize(*parents)
       @canceled = false
       @canceled_reason = nil
-      @canceled_mutex = Mutex.new
+      @canceled_mutex = Workflow::Unsafe.illegal_call_tracing_disabled { Mutex.new }
```

Comment:
Why not just use the workflow mutex here? Is it also used from non-workflow places?

Reply:
Yes. This is a general purpose "cancellation token" that is also used for activities (which is why it's not in the `Temporalio::Workflow` module, but rather here in the top level one)